### PR TITLE
fix(amplify_auth_cognito): fixes auth null safety issues

### PR DESCRIPTION
**Pull request by haverchuck**
*Description of changes:*
Fixes null-safety related auth issues found during testing.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by fjnoyp — APPROVED** [review]

File: packages/amplify_auth_cognito/android/src/main/kotlin/com/amazonaws/amplify/amplify_auth_cognito/types/FlutterUpdateUserAttributeResult.kt [hunk 1]
```diff
@@ -20,19 +20,11 @@ import com.google.gson.Gson
...
-            "updateAttributeStep" to raw.nextStep.updateAttributeStep.toString(),
-            "additionalInfo" to Gson().toJson(raw.nextStep.additionalInfo),
-            "codeDeliveryDetails" to mapOf(
-                "destination" to (raw.nextStep.codeDeliveryDetails?.destination ?: ""),
-                "deliveryMedium" to (raw.nextStep.codeDeliveryDetails?.deliveryMedium?.name ?: ""),
-                "attributeName" to (raw.nextStep.codeDeliveryDetails?.attributeName ?: "")
-            )
-        )
-    }
+    val nextStep: Map<String, Any> = com.amazonaws.amplify.amplify_auth_cognito.setNextStep(
```
Comment (cshfang) [inline packages/amplify_auth_cognito/android/src/main/kotlin/com/amazonaws/amplify/amplify_auth_cognito/types/FlutterUpdateUserAttributeResult.kt:23]: Could we import package at the top instead?
```suggestion
    val nextStep: Map<String, Any> = setNextStep(
```

File: packages/amplify_auth_cognito/example/ios/unit_tests/amplify_auth_cognito_tests.swift [hunk 2]
```diff
@@ -1344,7 +1344,7 @@ class amplify_auth_cognito_tests: XCTestCase {
                 XCTAssertEqual(false, res.toJSON()["isSignedIn"] as? Bool)
                 XCTAssertEqual("testid", res.toJSON()["identityId"] as? String)
                 // userSub error will result in map with one 'error' key
-                XCTAssertEqual(1, (res.toJSON()["tokens"] as?  [String: String])!.count)
+                XCTAssertEqual(nil, (res.toJSON()["tokens"] as?  [String: String]))
```
Comment (cshfang) [inline packages/amplify_auth_cognito/example/ios/unit_tests/amplify_auth_cognito_tests.swift:1347]: Nit: Use `XCTAssertNil` instead?
```suggestion
                XCTAssertNil(res.toJSON()["tokens"] as? [String: String])
```

File: packages/amplify_auth_cognito/example/ios/unit_tests/amplify_auth_cognito_tests.swift [hunk 3]
```diff
@@ -1389,7 +1389,7 @@ class amplify_auth_cognito_tests: XCTestCase {
                 // all tokens should be present with userpool-only access
                 XCTAssertEqual(3, (res.toJSON()["tokens"] as?  [String: String])!.count)
                 // credentials map should be empty
-                XCTAssertEqual(0, (res.toJSON()["credentials"] as?  [String: String])!.count)
+                XCTAssertEqual(nil, (res.toJSON()["credentials"] as?  [String: String]))
```
Comment (cshfang) [inline packages/amplify_auth_cognito/example/ios/unit_tests/amplify_auth_cognito_tests.swift:1392]: Nit: Use XCTAssertNil instead?
```suggestion
                XCTAssertNil(res.toJSON()["credentials"] as? [String: String])
```

File: packages/amplify_auth_cognito/ios/Classes/FlutterUpdateUserAttributeResult.swift [hunk 4]
```diff
@@ -29,16 +29,22 @@ struct FlutterUpdateUserAttributeResult {
       self.additionalInfo = setAdditionalInfo(res: res)
       self.codeDeliveryDetails = setCodeDeliveryDetails(res: res)
     }
-
+
```
Comment (cshfang) [inline packages/amplify_auth_cognito/ios/Classes/FlutterUpdateUserAttributeResult.swift:32]: Nit
```suggestion

```

**Review by cshfang — COMMENTED** [review]
Left some comments but lgtm otherwise

**Review by Jordan-Nelson — COMMENTED** [review]
LGTM other than the comments @cshfang already left

File: packages/amplify_auth_cognito/android/src/main/kotlin/com/amazonaws/amplify/amplify_auth_cognito/types/FlutterUpdateUserAttributeResult.kt (hunk 1 above)
Comment (haverchuck) [reply packages/amplify_auth_cognito/android/src/main/kotlin/com/amazonaws/amplify/amplify_auth_cognito/types/FlutterUpdateUserAttributeResult.kt:23]: Done

File: packages/amplify_auth_cognito/example/ios/unit_tests/amplify_auth_cognito_tests.swift (hunk 2 above)
Comment (haverchuck) [reply packages/amplify_auth_cognito/example/ios/unit_tests/amplify_auth_cognito_tests.swift:1347]: Done!

File: packages/amplify_auth_cognito/ios/Classes/FlutterUpdateUserAttributeResult.swift (hunk 4 above)
Comment (haverchuck) [reply packages/amplify_auth_cognito/ios/Classes/FlutterUpdateUserAttributeResult.swift]: Fixed

**Review by Jordan-Nelson — APPROVED** [review]

**Review by cshfang — APPROVED** [review]

**Pull request merged**